lib/fastlane/plugin/wpmreleasetoolkit/actions/common/buildkite_trigger_build_action.rb

```diff
@@ -0,0 +1,90 @@
+module Fastlane
+  module Actions
+    class BuildkiteTriggerBuildAction < Action
+      def self.run(params)
+        require 'buildkit'
+
+        UI.message "Triggering build on branch #{params[:branch]} at #{params[:commit]}"
+
+        pipeline_name = {
+          PIPELINE: params[:pipeline_file]
```

Comment:
Update: It does not.

![image](https://user-images.githubusercontent.com/1218433/152171614-fb06339b-86a0-403a-bb7c-2191de0fe01d.png)

![image](https://user-images.githubusercontent.com/1218433/152171625-10ae74f6-7a51-4cdf-a8f6-bc944ddc7911.png)

In hindsight, that makes total sense! Buildkit is a wrapper around the Buildkite APIs, for it to expect to be called from a file system location that includes the pipeline file it's asked to run would be reductive.

This actually makes me feel better about my earlier remarks about wrapper wrapping wrappers. Given the different of purpose, we can specialize our own wrapper to add those expectations and make it worthwhile.

I think it would be fair for our action to enforce that the pipeline file exists. Fastlane actions are most usually used within the context of a single project.

Reply:
Update: It does not.

![image](https://user-images.githubusercontent.com/1218433/152171614-fb06339b-86a0-403a-bb7c-2191de0fe01d.png)

![image](https://user-images.githubusercontent.com/1218433/152171625-10ae74f6-7a51-4cdf-a8f6-bc944ddc7911.png)

In hindsight, that makes total sense! Buildkit is a wrapper around the Buildkite APIs, for it to expect to be called from a file system location that includes the pipeline file it's asked to run would be reductive.

This actually makes me feel better about my earlier remarks about wrapper wrapping wrappers. Given the different of purpose, we can specialize our own wrapper to add those expectations and make it worthwhile.

I think it would be fair for our action to enforce that the pipeline file exists. Fastlane actions are most usually used within the context of a single project.

Reply:
> I think it would be fair for our action to enforce that the pipeline file exists. Fastlane actions are most usually used within the context of a single project.

I guess there's two routes we can take:

1. We could work on this as a super-flexible Fastlane-Buildkite integration to possibly extract as its own plugin. If we do that, then the check for the pipeline file existence in the local folder might limit the flexibility.
2. We could double down on using it constrained to a single project. That wouldn't preclude us from bundling as a dedicated action, but might make it less appealing (?) because of its assumption. In that case, IMHO, checking the pipeline file existence would be appropriate.

Reply:
So...this is tricky, because this is an implementation detail that's specific to how we do things. To the best of my knowledge, there's no global way to say "hey Buildkite, run this specific pipeline", so instead, our default project configuration does `buildkite-agent pipeline upload ./.buildkite/${PIPELINE:-pipeline.yml}`, so the `PIPELINE` environment variable determines what's run, if it's present.

We could try encoding this into the action, but then we'd need to replicate Buildkite's search structure (it searches the project root, the `.buildkite` directory, and possibly other places) in order to validate this. 

While I like the idea of helping the developer with this, I wonder if we're over-replicating work Buildkite already does (after all, IMHO the error message Buildkite emits that you posted above is fairly descriptive of the problem, and makes it pretty easy to see where you went wrong).

That said, I don't feel super strongly about it either, so interested to hear your thoughts 😄

Reply:
> So...this is tricky, because this is an implementation detail that's specific to how we do things. To the best of my knowledge, there's no global way to say "hey Buildkite, run this specific pipeline", so instead, our default project configuration does `buildkite-agent pipeline upload ./.buildkite/${PIPELINE:-pipeline.yml}`, so the `PIPELINE` environment variable determines what's run, if it's present.

How. Did. I. Miss. That?!

Okay cool, then that solves it for me. This is leaning way more towards a custom bit of tooling, for which we don't need to worry as much about flexibility but we can optimize to work well for us.

> We could try encoding this into the action, but then we'd need to replicate Buildkite's search structure (it searches the project root, the .buildkite directory, and possibly other places) in order to validate this.

Does it? Don't _we_ pass an exact path to `buildkite-agent pipeline upload`?

If my assumption is correct, we _could_ update our command in the Buildkite UI to be like:

```sh
buildkite-agent pipeline upload ${PIPELINE:-./.buildkite/pipeline.yml}
```

And then pass a proper path there.

But... I think, in particular because of the "internal tool" ramble above, that we can easily live with `pipeline_file` backing in certain assumptions specific to our coding standards and maybe compensate for it in the docs.

Reply:
Yeah, in this case _we_ have a specific way of doing it, but IIRC Pocket Casts is one example of a non-standard setup (non-standard == "not like our other projects"). If we were going to add support for _any_ Buildkite project we'd need to handle all of the file paths listed on https://buildkite.com/docs/agent/v3/cli-pipeline#uploading-pipelines and that seems like a lot of work.

I think we are in agreement! 😃

Reply:
> but IIRC Pocket Casts is one example of a non-standard setup (non-standard == "not like our other projects")

I updated Pocket Casts to follow the Automattic way™ not long ago 🤓 👌

Reply:
Ah! I was actually wondering how passing the `PIPELINE` as parameter here was working with the `pipeline-upload` step (which is hardcoded in the Buildkite project's web UI' settings, not in files in the git repo) and by what magic would Buildkite use this `PIPELINE` var.

I initially first thought that this was kind of a built-in an magic variable that Buildkite would know about and would in such case skip the `pipeline-upload` step defined in web settings and use the custom pipeline instead, which felt a bit too magical. The fact that it's all us explicitly using `${PIPELINE:-pipeline.yml}` makes way more sense, and I like that because it's way more explicit as well.

I completely missed that tbh. (I'm not sure if it you recently updated the pipeline setting on the web UI to change hardcoded `pipeline.yml` into `${PIPELINE:-pipeline.yml}` in order to support that PR, or if it was already there before?) A small nit I'd suggest then would be to name this env var something more explicit not that I understand that we do have full control over it, like `BUILDKITE_PIPELINE_FILE` rather than just `PIPELINE`?